pkg/reconciler/tenancy/clusterworkspacedeletion/clusterworkspace_deletion_controller.go

```diff
@@ -248,9 +259,19 @@ func (c *Controller) finalizeWorkspace(ctx context.Context, workspace *tenancyv1
 		if workspace.Finalizers[i] == deletion.WorkspaceFinalizer {
 			workspace.Finalizers = append(workspace.Finalizers[:i], workspace.Finalizers[i+1:]...)
 
+			lcn := logicalcluster.From(workspace)
+			listOpts := metav1.ListOptions{
+				LabelSelector: helper.WorkspaceLabelSelector(workspace.Name),
+			}
+			if err := c.kubeClusterClient.Cluster(lcn).RbacV1().ClusterRoles().DeleteCollection(ctx, backgroudDeletion, listOpts); err != nil && !kerrors.IsNotFound(err) {
+				return fmt.Errorf("could not delete clusterroles: %w", err)
+			}
+			if err := c.kubeClusterClient.Cluster(lcn).RbacV1().ClusterRoleBindings().DeleteCollection(ctx, backgroudDeletion, listOpts); err != nil && !kerrors.IsNotFound(err) {
+				return fmt.Errorf("could not delete clusterrolebindings: %w", err)
```

Comment:
Notable difference from the Workspace delete handler is we actually return errors now if we fail to delete and it is not due to CRs / CRBs not being found. This makes more sense in the context of a controller than in a typical REST handler IMO.